### PR TITLE
Remove the `listagg` macro for Spark

### PR DESCRIPTION
**Pull request by b-per**
This is a:
- [ ] bug fix PR with no breaking changes
- [x] new functionality

## Description & motivation

Following up the chat from [here](https://github.com/dbt-labs/dbt-project-evaluator/pull/214#issuecomment-1259900783), those changes allow us to remove `spark__listagg` from our repo.

Currently, though, we get some warnings with the message `order_by_clause is not supported for listagg on Spark/Databricks`, as we are not relying on the `order_by_clause` for Spark but it is used for other Warehouse.

## Checklist
- [x] I have verified that these changes work locally on the following warehouses (Note: it's okay if you do not have access to all warehouses, this helps us understand what has been covered)
    - [ ] BigQuery
    - [x] Postgres
    - [ ] Redshift
    - [ ] Snowflake
    - [x] DuckDB
    - [x] Databricks
- [ ] I have updated the README.md (if applicable)
- [ ] I have added tests & descriptions to my models (and macros if applicable)
- [ ] I have added an entry to CHANGELOG.md

### CODE REVIEW

**Comment by b-per** [comment]
The last commits remove `order_by_clause` for Spark, databricks and DuckDB and add a warning on the DuckDB side.

This handles the previous comment:
> Currently, though, we get some warnings with the message order_by_clause is not supported for listagg on Spark/Databricks, as we are not relying on the order_by_clause for Spark but it is used for other Warehouse.

File: models/marts/dag/fct_multiple_sources_joined.sql [hunk 1]
```diff
@@ -6,12 +6,17 @@ with direct_source_relationships as (
...
 ),
 
 multiple_sources_joined as (
     select
         child,
-        {{ dbt.listagg(measure='parent', delimiter_text="', '", order_by_clause='order by parent') }} as source_parents
+        {{ dbt.listagg(
+            measure='parent', 
+            delimiter_text="', '", 
+            order_by_clause='order by parent' if target.type not in ['databricks','duckdb','spark'])
```
Comment (dave-connors-3) [inline models/marts/dag/fct_multiple_sources_joined.sql:18]: do we need this at all if we're ordering in the previous CTE?

File: models/marts/dag/fct_source_fanout.sql [hunk 2]
```diff
@@ -6,12 +6,17 @@ with direct_source_relationships as (
...
 ),
 
 source_fanout as (
     select
         parent,
-        {{ dbt.listagg(measure='child', delimiter_text="', '", order_by_clause='order by child') }} as model_children
+        {{ dbt.listagg(
+            measure='child', 
+            delimiter_text="', '", 
+            order_by_clause='order by child' if target.type not in ['databricks','duckdb','spark'])
```
Comment (dave-connors-3) [inline models/marts/dag/fct_source_fanout.sql:18]: same comment as above

File: models/marts/structure/fct_model_naming_conventions.sql [hunk 3]
```diff
@@ -7,12 +7,17 @@ with all_graph_resources as (
...
 ), 
 
 appropriate_prefixes as (
     select 
         model_type, 
-        {{ dbt.listagg('prefix_value', "', '", 'order by prefix_value') }} as appropriate_prefixes
+        {{ dbt.listagg(
+            measure='prefix_value', 
+            delimiter_text="', '", 
+            order_by_clause='order by prefix_value' if target.type not in ['databricks','duckdb','spark'])
```
Comment (dave-connors-3) [inline models/marts/structure/fct_model_naming_conventions.sql:19]: thrice

File: models/marts/dag/fct_model_fanout.sql [hunk 4]
```diff
@@ -19,15 +19,28 @@ models_without_children as (
...
     select 
         all_dag_relationships.parent,
-        {{ dbt.listagg(measure='all_dag_relationships.child', delimiter_text="', '", order_by_clause='order by all_dag_relationships.child') }} as leaf_children
+        all_dag_relationships.child
     from all_dag_relationships
     inner join models_without_children
         on all_dag_relationships.child = models_without_children.parent
     where all_dag_relationships.distance = 1 and all_dag_relationships.child_resource_type = 'model'
+    group by 1, 2
+    order by 1, 2
```
Comment (graciegoheen) [inline models/marts/dag/fct_model_fanout.sql:28]: Can we just add a comment to all of these `order by`s explaining that they're necessary for the following `listagg` function

Comment (b-per) [reply models/marts/dag/fct_model_fanout.sql:28]: Can do

File: models/marts/dag/fct_multiple_sources_joined.sql (hunk 1 above)
Comment (b-per) [reply models/marts/dag/fct_multiple_sources_joined.sql:18]: Yes, because not all DW follow the order by of the previous data/CTE

Comment (dave-connors-3) [reply models/marts/dag/fct_multiple_sources_joined.sql:18]: oh weird -- so even though the `order by` is in the CTE, the result of the listagg may not respect that ordering?

**Comment by b-per** [comment]
Closes #205

**Review by dave-connors-3 — APPROVED** [review]
🦞 -- thanks for the follow up on the order by behavior! Pretty strange IMO that the ordering behaves differently. 

This is also duplicative of your open DuckDB PR, correct? Looks like this has all the stuff we would need for CI and materializations. If so, let's close out the other!

**Pull request merged**